File: services/gateways/bff/src/modules/interview/usecases/complete-interview.usecase.ts
import { Injectable } from "@nestjs/common";
import { InterviewGrpcService } from "../../../infra/grpc/services/interview-grpc.service";

export class CompleteInterviewCommand {
    constructor(public readonly interviewId: string) {}
}

@Injectable()
export class CompleteInterviewUseCase {
    constructor(private readonly interviewGrpcService: InterviewGrpcService) {}

    async execute(command: CompleteInterviewCommand) {
        const response = await this.interviewGrpcService.completeInterview(command.interviewId);

        return {
            interviewId: response.interviewId,
            status: response.status,
            endedAt: response.endedAt,
        };
    }
}

File: services/domains/interview/src/main/java/me/unbrdn/core/interview/application/port/in/CompleteInterviewUseCase.java
package me.unbrdn.core.interview.application.port.in;

import java.util.UUID;

/** Use Case for completing an interview */
public interface CompleteInterviewUseCase {
    CompleteInterviewResult execute(CompleteInterviewCommand command);

    record CompleteInterviewCommand(UUID interviewId) {}

    record CompleteInterviewResult(String interviewId, String status, String endedAt) {}
}
